tuf/ngclient/_internal/metadata_bundle.py

```diff
@@ -0,0 +1,461 @@
+                    )
+
+        if new_snapshot.signed.is_expired(self.reference_time):
+            raise exceptions.ExpiredMetadataError("New snapshot is expired")
+
+        self._bundle["snapshot"] = new_snapshot
+        logger.debug("Updated snapshot")
+
+    def update_targets(self, data: bytes):
+        """Verifies and loads 'data' as new top-level targets metadata.
+
+        Args:
+            data: unverified new targets metadata as bytes
+
+        Raises:
+            RepositoryError: Metadata failed to load or verify. The actual
+                error type and content will contain more details.
+        """
+        self.update_delegated_targets(data, "targets", "root")
+
+    def update_delegated_targets(
+        self, data: bytes, role_name: str, delegator_name: str
+    ):
+        """Verifies and loads 'data' as new metadata for target 'role_name'.
+
+        Args:
+            data: unverified new metadata as bytes
+            role_name: The role name of the new metadata
+            delegator_name: The name of the role delegating the new metadata
+
+        Raises:
+            RepositoryError: Metadata failed to load or verify. The actual
+                error type and content will contain more details.
+        """
+        if self.snapshot is None:
+            raise RuntimeError("Cannot load targets before snapshot")
+
+        delegator = self.get(delegator_name)
+        if delegator is None:
+            raise RuntimeError("Cannot load targets before delegator")
+
+        logger.debug("Updating %s delegated by %s", role_name, delegator_name)
+
+        # Verify against the hashes in snapshot, if any
+        meta = self.snapshot.signed.meta.get(f"{role_name}.json")
+        if meta is None:
+            raise exceptions.RepositoryError(
+                f"Snapshot does not contain information for '{role_name}'"
+            )
+
+        hashes = meta.hashes or {}
+        for algo, stored_hash in hashes.items():
+            digest_object = sslib_hash.digest(algo)
+            digest_object.update(data)
+            observed_hash = digest_object.hexdigest()
+            if observed_hash != stored_hash:
+                # TODO: Error should derive from RepositoryError
+                raise exceptions.BadHashError(stored_hash, observed_hash)
+
+        try:
+            new_delegate = Metadata.from_bytes(data)
+        except DeserializationError as e:
+            raise exceptions.RepositoryError("Failed to load snapshot") from e
+
+        if new_delegate.signed.type != "targets":
+            raise exceptions.RepositoryError(
+                f"Expected 'targets', got '{new_delegate.signed.type}'"
+            )
+
+        if not verify_with_threshold(delegator, role_name, new_delegate):
+            raise exceptions.UnsignedMetadataError(
+                f"New {role_name} is not signed by {delegator_name}",
+                new_delegate,
+            )
```

Comment:
this might be more readable if we called it `new_delegate_name` instead of `role_name` -- so the connection between the name and new_delegate metadata is clear.

Of course, if Metadata included the name (#1425), this would be even clearer:
```
if not verify_with_threshold(delegator, new_delegate):
    raise exceptions.UnsignedMetadataError(
        f"New {new_delegate.name} is not signed by {delegator.name}",
        new_delegate,
    )
```

Reply:
👍 completely agree with using variable names that make the connection clearer.